### ISSUE TITLE
SPSearchTopology doesn't work when NodeName is fully-qualified

### ISSUE THREAD

**Issue by joshtransient** [PowerShell/SharePointDsc#845]
**Details of the scenario you try and problem that is occurring:**
Creating a search topology does not work when NodeName is a fully-qualified domain name (e.g. _server.domain.com_ instead of simply _server_)

In MSFT_SearchTopology.psm1, lines 250-257:
````
$searchService = Get-SPEnterpriseSearchServiceInstance -Identity $searchServer `
                                                       -ErrorAction SilentlyContinue
if ($null -eq $searchService)
{
    $domain = (Get-CimInstance -ClassName Win32_ComputerSystem).Domain
    $searchServer = "$searchServer.$domain"
    $searchService = Get-SPEnterpriseSearchServiceInstance -Identity $searchServer
}
````

Can we add a check for FQDN somewhere here? Something like...
(edit: had the wrong operator)
````
if($searchServer -like '*.*') {
  $searchServer = $searchServer.Split('.')[0]
}
````

**The DSC configuration that is using the resource:**
The resource, when called:
````
SPSearchTopology SearchTopology {
      PsDscRunAsCredential    = $Node.SetupAccount
      ServiceAppName          = 'Global Search'
      Admin                   = $AllNodes.Where({ 'Admin'               -in $_.SearchRoles }).NodeName
      Crawler                 = $AllNodes.Where({ 'Crawler'             -in $_.SearchRoles }).NodeName
      ContentProcessing       = $AllNodes.Where({ 'ContentProcessing'   -in $_.SearchRoles }).NodeName
      AnalyticsProcessing     = $AllNodes.Where({ 'AnalyticsProcessing' -in $_.SearchRoles }).NodeName
      QueryProcessing         = $AllNodes.Where({ 'QueryProcessing'     -in $_.SearchRoles }).NodeName
      IndexPartition          = $AllNodes.Where({ 'Index' -in $_.SearchRoles -and $_.IndexPartition -eq 0 }).NodeName
      FirstPartitionDirectory = Join-Path -Path $Node.SearchIndexFolder -ChildPath '0'
      DependsOn               = @('[SPSearchServiceApp]SearchServiceApp','[SPMinRoleCompliance]MinRoleCompliance')
  }
````

Sanitized sample of my PSD1:
````
@{
  AllNodes = @(

    @{
      NodeName          = '*'
      SearchIndexFolder = 'D:\Data\Index'
    },

    @{
      NodeName        = 'server.subdomain.domain.com'
      ServerRole      = 'ApplicationWithSearch'
      RunCentralAdmin = $true
      SearchRoles     = @('Admin','Crawler','ContentProcessing','AnalyticsProcessing','QueryProcessing','Index')
      IndexPartition  = 0
    },
}
````

**Version of the Operating System and PowerShell the DSC Target Node is running:**
Server 2012 R2, Windows PowerShell 5.1

**Version of SharePoint that is used (e.g. SharePoint 2016):**
SharePoint 2016, FP2

**Version of the DSC module you're using:**
2.3.0.0

**Comment by johlju** [comment]
Can you provide the verbose log output when you get an error using the FQDN? Obfuscate any sensitive information.

**Comment by joshtransient** [comment]
First instance of error starts at line 1455.

[SPSearchTopology-sanitized.log](https://github.com/PowerShell/SharePointDsc/files/2241139/SPSearchTopology-sanitized.log)

**Comment by johlju** [comment]
I leave it for someone better at the Search Service to see what's going on. But label this as a bug in the meantime.

**Comment by ykuijs** [comment]
Closed via PR #850
